src/main/java/com/rabbitmq/stream/ConsumerFlowControlStrategy.java

```diff
@@ -0,0 +1,49 @@
+package com.rabbitmq.stream;
+
+import com.rabbitmq.stream.impl.Client;
```

Comment:
Got it.

Instead of using the entire `Client`, I'll create an interface specifically for asking for credits and use that instead. This could hurt some implementations that want to do other things with the connection than just asking for credits, but we'll cross that bridge once we get to it.

The protocol changes could be treated by deprecating and redirecting the methods in a retro-compatible way. The `ConsumerFlowControlStrategy` interface is only public API because the factory needs to be public API, but noted, I'll find a way to avoid this when possible.

I have some non-finished code introducing the first non-default Flow Control strategy, after I commit that later I'll refactor these dependencies.

Reply:
Check it out later, I believe it's at least better now